### DISCUSSION TITLE
Is it possible to get "BleedBox" and "TrimBox" value of a page?

### DISCUSSION THREAD

**Discussion by mlove4u** [pymupdf/PyMuPDF#1177] [Q&A]
I have some pdf files created by Adobe InDesign (with bleed setting). And I want to crop only the "TrimBox" range of every page to image. Is it possible to get the coordinate?
I can get the info with code:
```
>>> doc = fitz.open("trimbox.pdf")
>>> page = doc.load_page(0)
>>> xref = page.xref
>>> doc.xref_get_key(xref, "TrimBox")
('array', '[36.8499 36.8499 632.126 878.74]')
>>> doc.xref_get_key(xref, "BleedBox")
('array', '[28.346 28.346 640.629 887.244]')
#
>>> page.mediabox
Rect(0.0, 0.0, 668.9760131835938, 915.5900268554688)
>>> page.cropbox
Rect(0.0, 0.0, 668.9760131835938, 915.5900268554688)
```
Is there more simple way to do this? Like:
```
page.trimbox
page.bleedbox
```
[trimbox.pdf](https://github.com/pymupdf/PyMuPDF/files/6893763/trimbox.pdf)

**Comment by JorjMcKie** [comment]
> Is there more simple way to do this? 

No, currently there is not. Do not forget to multiply the result by `page.transformation_matrix` to get MuPDF coordinates.
But you seem to intend setting the CropBox to the TrimBox anyway. You can therefore directly do this 4-liner:
```python
for page in doc:
    val = doc.xref_get_key(page.xref, "TrimBox")
    if val[0] != "null":
        doc.xref_set_key(page.xref, "CropBox", val[1])
```
This also tackles orchid cases, where the trimbox value is not given as a direct array, but indirectly (as an xref).
